File: seeker/snippet/telegram_atm_bot.py
//date: 2025-02-04T17:05:45Z
//url: https://api.github.com/gists/02981d86fcf5e6614c0ebf917a44949a
//owner: https://api.github.com/users/yadnyeshkolte

import telebot
import random
import sqlite3

class TelegramATMBot:
 "**********"  "**********"  "**********"  "**********"  "**********"d "**********"e "**********"f "**********"  "**********"_ "**********"_ "**********"i "**********"n "**********"i "**********"t "**********"_ "**********"_ "**********"( "**********"s "**********"e "**********"l "**********"f "**********", "**********"  "**********"b "**********"o "**********"t "**********"_ "**********"t "**********"o "**********"k "**********"e "**********"n "**********") "**********": "**********"
        self.bot = "**********"
        self.authorized_users = {}
        self.account_balance = 15000
        self.transactions = []
        self.denominations = [200, 100, 50]
        self.notes_left = [5, 10, 10]
        
        # Setup bot message handlers
        self.register_handlers()

    def register_handlers(self):
        @self.bot.message_handler(commands=['start'])
        def send_welcome(message):
            """Display welcome message and available commands"""
            welcome_text = """
            Welcome to ATM Bot! 
            Available Commands:
            /login - Login to ATM
            /withdraw - Withdraw money
            /deposit - Deposit money
            /balance - Check balance
            /transactions - View transaction history
            """
            self.bot.reply_to(message, welcome_text)

    def authenticate(self, user_id):
        """User authentication mechanism"""
        otp = random.randint(1000, 9999)
        # Send OTP via Telegram
        self.bot.send_message(user_id, f"Your OTP is: {otp}")
        return self.verify_otp(user_id, otp)

    def verify_otp(self, user_id, generated_otp):
        """OTP verification logic"""
        # Implement OTP verification mechanism
        pass

    def withdraw_money(self, amount):
        """Money withdrawal logic"""
        if amount % 100 != 0:
            return "Invalid amount. Must be multiple of 100."
        
        if amount > self.account_balance:
            return "Insufficient funds."
        
        # Implement note dispensing logic
        temp_amount = amount
        for i, denomination in enumerate(self.denominations):
            while self.notes_left[i] > 0 and temp_amount >= denomination:
                temp_amount -= denomination
                self.notes_left[i] -= 1
        
        self.account_balance -= amount
        self.transactions.append(amount)
        return f"Withdrawn {amount}. Remaining balance: {self.account_balance}"

    def deposit_money(self, amount):
        """Money deposit logic"""
        if amount % 100 != 0:
            return "Invalid amount. Must be multiple of 100."
        
        self.account_balance += amount
        self.transactions.append(amount)
        return f"Deposited {amount}. New balance: {self.account_balance}"

    def check_balance(self):
        """Balance inquiry"""
        return f"Current Balance: {self.account_balance}"

    def view_transactions(self):
        """View transaction history"""
        return "\n".join([f"Transaction {i+1}: {amt}" for i, amt in enumerate(self.transactions)])

# Bot initialization and running
bot_token = "**********"
atm_bot = "**********"
atm_bot.bot.polling()

File: seeker/snippet/TelegramATMBot.java
//date: 2025-02-04T17:05:45Z
//url: https://api.github.com/gists/02981d86fcf5e6614c0ebf917a44949a
//owner: https://api.github.com/users/yadnyeshkolte

import org.telegram.telegrambots.bots.TelegramLongPollingBot;
import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
import org.telegram.telegrambots.meta.api.objects.Update;
import java.util.ArrayList;
import java.util.Random;

public class TelegramATMBot extends TelegramLongPollingBot {
    private static final String BOT_TOKEN = "**********"
    private static final String BOT_USERNAME = "your_bot_username";

    // Account Management Variables
    private double accountBalance = 15000.0;
    private List<Double> transactions = new ArrayList<>();
    private int[] denominations = {200, 100, 50};
    private int[] notesLeft = {5, 10, 10};
    private boolean isAuthorized = false;

    @Override
    public void onUpdateReceived(Update update) {
        if (update.hasMessage() && update.getMessage().hasText()) {
            String messageText = update.getMessage().getText();
            long chatId = update.getMessage().getChatId();

            switch (messageText) {
                case "/start":
                    sendWelcomeMessage(chatId);
                    break;
                case "/login":
                    handleLogin(chatId);
                    break;
                case "/withdraw":
                    handleWithdraw(chatId, update);
                    break;
                case "/deposit":
                    handleDeposit(chatId, update);
                    break;
                case "/balance":
                    checkBalance(chatId);
                    break;
                case "/transactions":
                    viewTransactions(chatId);
                    break;
            }
        }
    }

    private void sendWelcomeMessage(long chatId) {
        String welcome = "Welcome to ATM Bot!\n" +
                         "Available Commands:\n" +
                         "/login - Authenticate\n" +
                         "/withdraw - Withdraw Money\n" +
                         "/deposit - Deposit Money\n" +
                         "/balance - Check Balance\n" +
                         "/transactions - View History";
        sendMessage(chatId, welcome);
    }

    private void handleLogin(long chatId) {
        // Generate OTP
        int otp = generateOTP();
        sendMessage(chatId, "Your OTP is: " + otp);
        // Implement OTP verification mechanism
    }

    private int generateOTP() {
        return new Random().nextInt(9000) + 1000;
    }

    private void handleWithdraw(long chatId, Update update) {
        if (!isAuthorized) {
            sendMessage(chatId, "Please login first!");
            return;
        }

        // Implement amount input and validation
        double amount = extractAmount(update);
        
        if (amount % 100 != 0) {
            sendMessage(chatId, "Invalid amount. Must be multiple of 100.");
            return;
        }

        if (amount > accountBalance) {
            sendMessage(chatId, "Insufficient funds!");
            return;
        }

        // Process withdrawal
        processWithdrawal(chatId, amount);
    }

    private void processWithdrawal(long chatId, double amount) {
        accountBalance -= amount;
        transactions.add(-amount);
        
        // Note dispensing logic
        double remainingAmount = amount;
        for (int i = 0; i < denominations.length; i++) {
            while (notesLeft[i] > 0 && remainingAmount >= denominations[i]) {
                remainingAmount -= denominations[i];
                notesLeft[i]--;
            }
        }

        sendMessage(chatId, "Withdrawn: " + amount + 
                    "\nRemaining Balance: " + accountBalance);
    }

    private void handleDeposit(long chatId, Update update) {
        if (!isAuthorized) {
            sendMessage(chatId, "Please login first!");
            return;
        }

        double amount = extractAmount(update);
        
        if (amount % 100 != 0) {
            sendMessage(chatId, "Invalid amount. Must be multiple of 100.");
            return;
        }

        accountBalance += amount;
        transactions.add(amount);
        sendMessage(chatId, "Deposited: " + amount + 
                    "\nNew Balance: " + accountBalance);
    }

    private void checkBalance(long chatId) {
        if (!isAuthorized) {
            sendMessage(chatId, "Please login first!");
            return;
        }
        sendMessage(chatId, "Current Balance: " + accountBalance);
    }

    private void viewTransactions(long chatId) {
        if (!isAuthorized) {
            sendMessage(chatId, "Please login first!");
            return;
        }
        
        StringBuilder transactionHistory = new StringBuilder("Transaction History:\n");
        for (int i = 0; i < transactions.size(); i++) {
            transactionHistory.append("#")
                              .append(i + 1)
                              .append(": ")
                              .append(transactions.get(i))
                              .append("\n");
        }
        sendMessage(chatId, transactionHistory.toString());
    }

    private void sendMessage(long chatId, String text) {
        SendMessage message = new SendMessage();
        message.setChatId(String.valueOf(chatId));
        message.setText(text);
        try {
            execute(message);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public String getBotUsername() {
        return BOT_USERNAME;
    }

    @Override
    public String getBotToken() {
        return BOT_TOKEN;
    }
}  }
}